Add minimal VS2022 support.

hbono opened the pull request:
Greetings,

I have updated NuGet packages of your extension and resolved their packages conflicts so your extension can work well on VS2022. (In brief, this commit is a quick fix for Issue #6.)
Would it be possible to merge this commit and update your extension?

Regards,

Hironori Bono

hbono committed:
Add minimal VS2022 support.
This change updates NuGet packages and resolves packages conflicts so this extension can work well on VS2022. (This change does not update package manifests because this commit may break backward compatibility.)

Smurf-IV commented:
@hbono I built your project from your code base, and both your `ready` and bin\release versions give this with VS 17.3.3
![image](https://user-images.githubusercontent.com/2418812/190579449-deafc393-0ec5-45c5-81f2-91ad14adaa87.png)

Note: you do not have an issues on your GitHub project to report this.

hbono committed:
Update the package manifest.
This change updates the `source.extension.vsixmanifest` file to add VS2022 support. (This change does NOT include ARM64 support because I do not have ARM PCs.)

hbono commented:
@Smurf-IV

Sorry for my slow response. (I do not check GitHub notifications so often.)
Oops, I forgot including a change to the `extension.vsixmanifest` file. I have updated this pull request to include it. I would note that this update does NOT include ARM64 support because I cannot verify this extension works on ARM64 PCs. (This project only uses C#. So, I suppose this extension works on ARM64 PCs as well.)
Please feel free to send emails to me if you need to ping me. (My email address is included in the commit logs.)

Regards,

Hironori Bono

DevX-Cipher commented:
this would help if it builds

hbono commented:
Hello,

Thank you for your feedback.
Fortunately or unfortunately, I could build my branch when I built it using [Visual Studio 2022 version 17.10.1](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#17101--visual-studio-2022-version-17101). Would it be possible to give me errors if you have problems? (Also, please feel free to ask me if you have difficulties in building my branch.)

Regards,

Hironori Bono

DevX-Cipher commented:
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error		The target "IncludeVSIXItems" does not exist in the project.	ClearRecent

DevX-Cipher commented:
@hbono i fixed the issue it was missing packages you don't get clear all form the start up page of visual studio also when you clear it inside of visual studio it removes all pin projects

DevX-Cipher commented:
@hbono also adding the option in the file menu to pinned current project opened as well

DevX-Cipher commented:
@hbono did you end finding the solution for pinned projects yet or no

hbono commented:
@modz2014 

Sorry for my slow response.
When I select "File -> Recent Projects and Solutions -> Clear all" on [Visual Studio 17.11.4](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#version-17114), ClearRecent deletes the projects pinned in its "Start Window". Am I missing something?

Regards,

Hironori Bono
![StartWindow](https://github.com/user-attachments/assets/3367cb85-aa0d-407d-8275-64a4c2136ec0)

DevX-Cipher commented:
we need to stop it from removing pinned ones

hbono commented:
@modz2014 

Thank you for your response.
Unfortunately, it is not a scope of this pull list. This pull list is just for **supporting Visual Studio 2022** as written in its title. I will investigate how to exclude pinned projects and create another pull list some time. (Unfortunately, I do not have enough time to do it for now.)

Regards,

Hironori Bono

DevX-Cipher commented:
i tried searching it up and can not find any documents about it also it would be nice to exclude pinned project a lot of people are asking about it

DevX-Cipher commented:
@hbono have you come back to this yet

DevX-Cipher commented:
@hbono the only way to check if the project is pinned is by looking in ApplicationPrivateSettings.xml for IsFavorite:true or false if not pinned
